astronomer/airflow/version_check/update_checks.py

```diff
@@ -381,13 +397,69 @@ def available_update(self):
                 "app_name": "Astronomer Runtime",
             }
 
+        return update_notice
+
+    def get_eol_notice(self, current_version) -> dict[str, Any] | None:
+        """
+        Get the EOL notice information if the current version is near or past its EOL.
+
+        :param current_version: The current runtime version information.
+        """
+        if current_version and current_version.end_of_support:
+            now = utcnow()
+            days_to_eol = (current_version.end_of_support - now).days
+            eol_level = (
+                'critical'
+                if days_to_eol <= 0
+                else 'warning'
+                if days_to_eol <= self.eol_warning_threshold_days
+                else ''
+            )
+            return {
+                "level": eol_level,
+                "version": current_version.version,
+                "app_name": "Astronomer Runtime",
+                "days_to_eol": days_to_eol,
+                "dismissed_until": current_version.eos_dismissed_until,
+            }
         return None
 
+    def available_update(self) -> dict[str, dict[str, Any] | None]:
+        """Check for available updates and EOL notices for the current runtime version."""
+        from .plugin import eol_warning_opt_out
+        from .models import AstronomerAvailableVersion
+
+        session = get_state(app=current_app).db.session
+        available_releases = session.query(AstronomerAvailableVersion).filter(
+            AstronomerAvailableVersion.hidden_from_ui.is_(False)
+        )
+
+        runtime_version = version.parse(get_runtime_version())
+        base_version = runtime_version.major
+
+        sorted_releases = sorted(available_releases, key=lambda v: version.parse(v.version), reverse=True)
+
+        update_notice = self.get_update_notice(sorted_releases, runtime_version, base_version)
+
+        eol_notice = None
+        if not eol_warning_opt_out:
+            current_version = (
+                session.query(AstronomerAvailableVersion)
+                .filter(AstronomerAvailableVersion.version == str(runtime_version))
+                .one_or_none()
+            )
+            eol_notice = self.get_eol_notice(current_version)
+
+        return {"update_notice": update_notice, "eol_notice": eol_notice}
+
     def new_template_vars(self):
         return {
             # Fetch it once per template render, not each time it's accessed
             'cea_update_available': lazy_object_proxy.Proxy(self.available_update),
```

Comment:
I still think we should rename this template var. Or, leave this one for update notices and add another for eol.

Reply:
I have made separate method for EOL notice and updates now.